src/components/Table/index.tsx

```diff
@@ -0,0 +1,306 @@
+  hasChildren: hasChildrenFunc,
+  identifierField: ID,
+  linkerField: ID,
+  parentIdentifierField: PARENT_ID,
+  rootParentId: ROOT_PARENT_ID,
+};
+
+function DrillDownTable<D extends object>(props: DrillDownTableProps<D>) {
+  const { columns, data, parentIdentifierField, hasChildren } = props;
+  const parentNodes =
+    data && parentIdentifierField ? data.map((el: Dictionary) => el[parentIdentifierField]) : [];
+  const [currentParentId, setCurrentParentId] = useState<string>(props.rootParentId);
+  const [pageData, setpageData] = useState<D[]>([]);
+  const [pgCount, setPageCount] = useState<number>(0);
+  const mutatedColumns = React.useMemo(() => columns.map(mutateColumns), []) as Array<
+    ColumnInstance<D>
+  >;
+
+  const fetchData = React.useCallback(
+    ({ pageSize, pageIndex, skipPageResetRef, currentParentId: parentId }) => {
+      skipPageResetRef.current = true;
+      const startRow = pageSize * pageIndex;
+      // TODO - enforce typings for the above parameters
+      // tslint:disable-next-line: restrict-plus-operands
+      const endRow = startRow + pageSize;
+      const filterByLevel = props.data.filter((row: any) => {
+        return row[parentIdentifierField] === parentId;
+      });
+
+      const filterByPage = filterByLevel.slice(startRow, endRow);
+
+      setpageData(filterByPage);
+      setPageCount(Math.ceil(filterByLevel.length / pageSize));
+    },
+    []
+  );
+
+  /** Get modified columns
+   * Modify the linker column to include an indicator that you can use to
+   * drill-down
+   */
+  function mutateColumns(el: Dictionary) {
+    const { linkerField, CellComponent, extraCellProps } = props;
+    if (el.hasOwnProperty('columns') && el.columns && el.columns.length > 0) {
+      const newColumns = el.columns.map(mutateColumns);
+      el.columns = newColumns;
+    }
+
+    if (el.accessor === linkerField) {
+      el.Cell = (cell: Cell) => {
+        if (CellComponent !== undefined) {
+          const { identifierField } = props;
+
+          let thisCellHasChildren = false;
+          if (hasChildren && identifierField && hasChildren(cell, parentNodes, identifierField)) {
+            thisCellHasChildren = true;
+          }
+
+          const cellProps: DropDownCellProps = {
+            cell,
+            cellValue: cell.value,
+            hasChildren: thisCellHasChildren,
+          };
+
+          if (extraCellProps !== undefined) {
+            Object.assign(cellProps, extraCellProps);
+          }
+          return <CellComponent {...cellProps} />;
+        }
+        return cell.value;
+      };
+    }
+    return el;
+  }
+
+  return (
+    <Table
+      columns={mutatedColumns}
+      data={pageData}
+      pgCount={pgCount}
+      fetchData={fetchData}
+      currentParentId={currentParentId}
+      setCurrentParentId={setCurrentParentId}
+      parentNodes={parentNodes}
+    />
+  );
+}
+
+DrillDownTable.defaultProps = defautlDrillDownTableProps;
+
+export { DrillDownTable };
```

Comment:
Cool, let's export everything else be declared in this file here as well 🙏

Reply:
cool